crypto/bn/bn_prime.c

```diff
@@ -19,14 +19,48 @@
  */
 #include "bn_prime.h"
 
-static int witness(BIGNUM *w, const BIGNUM *a, const BIGNUM *a1,
-                   const BIGNUM *a1_odd, int k, BN_CTX *ctx,
-                   BN_MONT_CTX *mont);
 static int probable_prime(BIGNUM *rnd, int bits, prime_t *mods);
 static int probable_prime_dh_safe(BIGNUM *rnd, int bits,
                                   const BIGNUM *add, const BIGNUM *rem,
                                   BN_CTX *ctx);
 
+#if BN_BITS2 == 64
+#define BN_DEF(l, h) 0x##h##l##ULL
```

Comment:
We have been burnt by this before. I mean by 0x##h##l thing. See RT#3165 at rt.openssl.org. ULL can also be problematic for some overzealous compilers. `(BN_ULONG)hi<<32|lo` is preferred.

Reply:
> `(BN_ULONG)hi<<32|lo` is preferred.

It naturally means that "anatomically correct" hexadecimals are passed, i.e. 0xABC and not ABC.

Reply:
Will change (ULL is used in bn_nist.c)